### DISCUSSION TITLE
Vi må slutte med props for ting som "Settes automatisk av ..."

### DISCUSSION THREAD

**Discussion by joms** [fremtind/jokul#2791] [Annet]
En relativt bastant påstand, men etter å ha tukla rundt med `ExpandableTableRow` har jeg nå forvirret meg selv mer enn én gang med mitt eget API. Spesielt i `ExpandableTableRowController` med sine props:

```typescript
export interface ExpandableTableRowControllerProps extends TableCellProps {
    /** Settes automatisk av ExpandableTableRow */
    isOpen?: boolean;
    onClick?: () => void;
}
```

Her er APIet forvirrende når jeg skal bruke komponenten ettersom VSCode sier at her er det props jeg kan bruke, og selv om kommentaren også blir med er det lett å overse denne. Dette er et mønster vi bruker på flere komponenter i Jøkul. Spesielt er dette komponenter med mer kompleks funksjonalitet hvor det finnes en mor-komponent som styrer et child.

Som et alternativ til dette mønsteret burde vi heller holde sånne verdier med Context. Selv om dette kompliserer koden noe, isolerer vi props som ikke skal eksponeres og unngår forvirring i bruken av props.

Snurr debatt!

**Comment by wkillerud** [comment]
Enig i at det er en litt "uting", og at context bør være hovedregelen 👍 Har syndet her selv, spesielt når det er en direkte parent->child, men det kan skape forvirring som du sier.

**Comment by piofinn** [comment]
Enig! Hvis komponenten kun fungerer _i kontekst av_ info fra en annen komponent er det neppe feil å putte det i en `context` 😬

**Comment by henrikhermansen** [comment]
Er det slik at dette er typer som synes og brukes av konsumenter av komponentene? I så fall er jeg enig at det er fint å unngå å eksponere typer som ikke egentlig er meningen å bruke.

**Reply by wkillerud** [reply]
Stemmer, de synes f. eks. i Intellisense og lignede

**Comment by joms** [comment]
Da tenker jeg det er generell enighet om at vi gjør dette og lukker diskusjonen 👍
